woocommerce/woocommerce
# Analyzer CLI: Add Database scanning

### All Submissions:






### Changes proposed in this Pull Request:

This PR adds database schema and update scanning to the Analyzer CLI command tool.

Schemas between branches being compared are evaluated for equality and database update functions are scanned for.

### How to test the changes in this Pull Request:

1. Make sure your local `trunk` is up to date.
2. Introduce a schema change somewhere in this string (on this branch)

https://github.com/woocommerce/woocommerce/blob/7e49c151b74a6d82bc43e57fbbba29db7c4dfc71/plugins/woocommerce/includes/class-wc-install.php#L1013-L1015

3. Introduce a database update function by adding a function this array

https://github.com/woocommerce/woocommerce/blob/7e49c151b74a6d82bc43e57fbbba29db7c4dfc71/plugins/woocommerce/includes/class-wc-install.php#L36

4. Commit your changes locally.
5. Run the analyzer
```
./tools/code-analyzer/bin/dev analyzer add/database-scanner-cli
```
6. See resulting notices

```
## SCHEMA CHANGES
---------------------------------------------------
 NOTICE | Schema changes detected in 6.6.0
---------------------------------------------------

## DATABASE UPDATES
---------------------------------------------------
 NOTICE | Database update found | wc_update_652_approved_things introduced in 6.5.2
---------------------------------------------------
```

### Other information:




### FOR PR REVIEWER ONLY:
